### PR TITLE
Add a test for the empty list type

### PR DESCRIPTION
**Pull request by yutopp**

### CODE REVIEW

File: test/test_empty_list_type.ml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  [%expect {|
+    (Ok (
+      AbstractCode (
+        ModDecl (
+          (AttrFile 1 test_empty_list_type.erl 1)
+          (AttrMod 1 test_empty_list_type)
+          (AttrExportType 3 (
+            (t 0)
+            (u 0)))
+          (DeclType 6 t () (TyPredef 6 nil ()))
```
Comment (amutake) [inline test/test_empty_list_type.ml]: memo: 
In [the document](http://erlang.org/doc/apps/erts/absform.html), 
> the empty list type [] cannot be distinguished from the predefined type nil().

so `[]` type can be considered as predefined `nil()` type.
